packages/backend/src/models/Membership.ts

```diff
@@ -99,19 +101,19 @@ export class Membership {
           (memberStart.getTime() >= startDate.getTime() &&
             memberStart.getTime() <= endDate.getTime())
         ) {
-          return false;
+          return true;
         }
       });
     }
 
-    return true;
+    return false;
   }
 
   public static async validate(data: Membership): Promise<APIError | void> {
-    if (await Membership.endDateAfterStartDate(data))
+    if (Membership.endDateAfterStartDate(data))
       return new APIError(400, 'Start date mustn\'t be after end date');
 
-    if (await Membership.onlyOneMembershipByMember(data))
+    if (await Membership.alreadyOnMembership(data))
```

Comment:
Don't put await in an if compartor, store the result in a separate variable/constant